### PR TITLE
Add duplicate search result API

### PR DESCRIPTION
**Pull request by sergey-misuk-valor**

### CODE REVIEW

File: src/hope_dedup_engine/apps/api/models/deduplication.py [hunk 1]
```diff
@@ -21,38 +20,34 @@ class State(models.IntegerChoices):
...
-    created_at = models.DateTimeField(_("created at"), auto_now_add=True)
-    #
-    # class Meta:
-    #     unique_together = ("deduplication_set", "filename")
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Duplicate(models.Model):
+    deduplication_set = models.ForeignKey(DeduplicationSet, on_delete=models.CASCADE)
+    first_reference_pk = models.IntegerField()
```
Comment (saxix) [inline src/hope_dedup_engine/apps/api/models/deduplication.py]: let's use CharField here. Remote systems can send anything

File: src/hope_dedup_engine/apps/api/models/deduplication.py [hunk 2]
```diff
@@ -21,38 +20,34 @@ class State(models.IntegerChoices):
...
+    first_filename = models.CharField(max_length=255)
+    second_reference_pk = models.IntegerField()
```
Comment (saxix) [inline src/hope_dedup_engine/apps/api/models/deduplication.py]: same as above

**Review by saxix — CHANGES_REQUESTED** [review]

File: src/hope_dedup_engine/apps/api/models/deduplication.py (hunk 1 above)
Comment (sergey-misuk-valor) [reply src/hope_dedup_engine/apps/api/models/deduplication.py]: Done

File: src/hope_dedup_engine/apps/api/models/deduplication.py (hunk 2 above)
Comment (sergey-misuk-valor) [reply src/hope_dedup_engine/apps/api/models/deduplication.py]: Done

**Review by saxix — APPROVED** [review]

**Pull request merged**